pkg/resource/notebook_instance_lifecycle_config/custom_delta.go

```diff
@@ -0,0 +1,72 @@
+package notebook_instance_lifecycle_config
+
+import (
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+)
+
+func modifyDeltaCreate(
+	delta *ackcompare.Delta,
+	a *resource,
+	b *resource,
+) *ackcompare.Delta {
+	if ackcompare.HasNilDifference(a.ko.Spec.OnCreate, b.ko.Spec.OnCreate) {
+		delta.Add("Spec.OnCreate", a.ko.Spec.OnCreate, b.ko.Spec.OnCreate)
+		return delta
+	}
+	//check length
+	if a.ko.Spec.OnCreate != nil && b.ko.Spec.OnCreate != nil {
+		if len(a.ko.Spec.OnCreate) != len(b.ko.Spec.OnCreate) {
+			delta.Add("Spec.OnCreate", a.ko.Spec.OnCreate, b.ko.Spec.OnCreate)
+			return delta
+		}
+	} else {
+		return delta
+	}
+	//Check variables, a and b have to be of equal length
+	onCreateLen := len(a.ko.Spec.OnCreate)
+	if a.ko.Spec.OnCreate != nil && b.ko.Spec.OnCreate != nil {
```

Comment:
is it possible to test this using the update test ? Else we should make sure to add it to unit tests.